src/Qoollo.PerformanceCounters/WinCounters/Counters/WinDeltaCounter.cs

```diff
@@ -0,0 +1,238 @@
+        {
+            var counterCpy = _winCounter;
+            if (counterCpy == null)
+                return Counter.FailureNum;
+
+            return counterCpy.Decrement();
+        }
+
+        /// <summary>
+        /// Уменьшить значение на "value"
+        /// </summary>
+        /// <param name="value">Значение, на которое уменьшаем</param>
+        /// <returns>Текущее значение или Counter.FailureNum</returns>
+        public override long DecrementBy(long value)
+        {
+            var counterCpy = _winCounter;
+            if (counterCpy == null)
+                return Counter.FailureNum;
+
+            return counterCpy.IncrementBy(-value);
+        }
+
+        /// <summary>
+        /// Увеличить значение на 1
+        /// </summary>
+        /// <returns>Текущее значение или Counter.FailureNum</returns>
+        public override long Increment()
+        {
+            var counterCpy = _winCounter;
+            if (counterCpy == null)
+                return Counter.FailureNum;
+
+            return counterCpy.Increment();
+        }
+
+        /// <summary>
+        /// Увеличить значение на "value"
+        /// </summary>
+        /// <param name="value">Значение, на которое увеличиваем</param>
+        /// <returns>Текущее значение или Counter.FailureNum</returns>
+        public override long IncrementBy(long value)
+        {
+            var counterCpy = _winCounter;
+            if (counterCpy == null)
+                return Counter.FailureNum;
+
+            return counterCpy.IncrementBy(value);
+        }
+
+        /// <summary>
+        /// Задать значение
+        /// </summary>
+        /// <param name="value">Новое значение</param>
+        public override void SetValue(long value)
+        {
+            var counterCpy = _winCounter;
+            if (counterCpy == null)
+                return;
+
+            counterCpy.RawValue = value;
+        }
+
+        /// <summary>
+        /// Получает текущее значение счетчика производительности
+        /// </summary>
+        public override long CurrentValue
+        {
+            get
+            {
+                var counterCpy = _winCounter;
+                if (counterCpy == null)
+                    return Counter.FailureNum;
+
+                long lastMeasuredValue = Interlocked.Read(ref _lastMeasuredValue);
+                long currentValue = counterCpy.RawValue;
+                return currentValue - lastMeasuredValue;
+            }
+        }
+
+        /// <summary>
+        /// Take next sample interval and return its value
+        /// </summary>
+        /// <returns>Difference between counter raw value during sample interval</returns>
+        public override long Measure()
+        {
+            var counterCpy = _winCounter;
+            if (counterCpy == null)
+                return FailureNum;
+
+            var curValue = (long) counterCpy.NextValue();
+            Interlocked.Exchange(ref _lastMeasuredValue, curValue);

```

Comment:
```NextValue()``` returns difference (delta) between two measurements. To calculate ```CurrentValue``` correctly it is required to store ```RawValue``` into ```_lastMeasuredValue``` field. And it is better to do this before the call to ```counterCpy.NextValue()``` to prevent ordering problems.